source/github-events/on-new-comments.ts

```diff
@@ -4,7 +4,17 @@ import delegate from 'delegate-it';
 const discussionsWithListeners = new WeakSet();
 const handlers = new Set<VoidFunction>();
 const delegates = new Set<delegate.Subscription>();
-const observer = new MutationObserver(run);
+
+let commentsCount = 0;
+const observer = new MutationObserver(() => {
+	const commentsNewCount = select.all('.js-comment').length;
+	if (commentsNewCount > commentsCount) {
+		commentsCount = commentsNewCount; // Update the count ASAP to avoid duplicate calls
+		run();
+	}
```

Comment:
The mutation observer is triggered by all kinds of things: new events, comments edited/deleted, reactions added/removed on the body, etc.

Since the mutation records are identical every time, the simplest way I've found to check for the addition of new comments is to keep track of their current number.

Tested with regular as well as review comments.